### ISSUE TITLE
Release StatsBase.jl v1.0

### ISSUE THREAD

**Issue by non-Jedi** [JuliaStats/StatsBase.jl#493]
Similar to https://github.com/JuliaStats/Distributions.jl/issues/880. What needs to happen for StatsBase to be declared v1.0? Are there additional planned breaking changes, or missing essential functions?

**Comment by nalimilan** [comment]
AFAIK the plan is rather to kill StatsBase before it reaches 1.0, by moving things to Statistics or to other more specialized packages.

**Comment by matbesancon** [comment]
Given that stdlib PRs are usually slower to get in, we should start working on that

**Comment by rofinn** [comment]
FWIW, I don't think moving functionality to Statistics prevents us from making a 1.0 release of StatsBase. If anything having a major release to identify and deprecate any potentially breaking changes while moving functionality to Statistics might be helpful.

**Comment by juliohm** [comment]
Fully support the move to Statistics. I will be super happy the day where we can just do `using Statistics` in our code and have all the basic building blocks in the scope (e.g. `mean`, `std`, `sample`) as opposed to separating pieces in there and `StatsBase`. 🙏

**Comment by tpapp** [comment]
I don't think that moving StatsBase to Statistics is a good idea before the latter is decoupled from Julia's release cycle.

StatsBase gets a release every few months, and PRs are merged without much of hassle, whereas nontrivial changes are costly to make for standard libraries. At the moment, moving to Statistics would also preclude any breaking changes, no matter how minor, until Julia 2.0.

**Comment by juliohm** [comment]
@tpapp are we still expecting major changes in the statistics API? If something is not solid yet in StatsBase.jl, it is probably because it deserves a separate package?

My main concern with the separation as it is now regards basic names like `mean, std, sample` being spread over different packages. Users would have a better experience by simply `using Statistics` and getting all these basic operations in the scope.

**Comment by tpapp** [comment]
APIs are quite difficult to get right. It's not that existing stuff in `StatsBase` is less solid than in other packages — it is certainly a mature package — it's just that having the option to change things without waiting for a major release of Julia is itself valuable.

It is very hard to draw the line for what's "basic" in statistics. IMO `Statistics` has the "super basic" stuff that is not even statistics for most people (eg a `mean`), `StatsBase` is the best place for well-established descriptive stats, non-parametric methods, and some [basic interface specs](https://juliastats.org/StatsBase.jl/stable/statmodels/#Abstraction-for-Statistical-Models-1), and other packages can do more specific things.

In any case, I think this is somewhat orthogonal to this issue (conditional on StatsBase not being merged into Statistics anytime soon). Tagging a 1.0 would be useful regardless.

**Comment by nalimilan** [comment]
Given that we haven't made lot of progress towards moving things to Statistics, I think we should tag 1.0 next time we want to make a breaking release. Let's mark issues with the `breaking` label to track these.

**Comment by brenhinkeller** [comment]
> At the moment, moving to Statistics would also preclude any breaking changes, no matter how minor, until Julia 2.0.

Is adding things considered breaking? If not, that may be more of a feature than a bug :p

But I guess going 1.0 in StatsBase doesn't preclude anything on that side (if anything signals that there is no need for further API changes to the functions therein, even if they get moved to Base)
